internal/node/node_loader.js

```diff
@@ -59,22 +59,18 @@ function resolveToModuleRoot(path) {
     throw new Error('resolveToModuleRoot missing path: ' + path);
   }
 
-  var match;
-  var lengthOfMatch = 0;
-  for (var i = 0; i < MODULE_ROOTS.length; i++) {
-    var m = MODULE_ROOTS[i];
-    var p = path.replace(m.module_name, m.module_root);
-    // Longest regex wins when multiple match
-    var len = m.module_name.toString().length;
-    if (p !== path && len > lengthOfMatch) {
-      lengthOfMatch = len;
-      match = p;
-    }
-  }
-  if (match) {
-    return match;
+  const orderedMatches = MODULE_ROOTS
+    // We want all possible matches.
+    .filter(m => m.module_name.test(path))
+    // Longest regex wins when multiple match.
+    .sort((a, b) => a.module_name.toString().length < b.module_name.toString().length);
```

Comment:
hmm, can we avoid sorting this list every time we resolve a module?

Reply:
Yes we can probably pre-sort them at the start.

Reply:
Done